File: 5_Longest_Palindromic_Substring/brute_force.py
class Solution:
    def check_palindromic(self, s:str) -> bool:
        n = len(s)
        for i in range(n // 2):
            if s[i] != s[n-i-1]:
                return False
        return True
    
    def longestPalindrome(self, s: str) -> str:
        max_len = 0
        
        for i in range(len(s)):
            for j in range(i+1, len(s)+1):
                if self.check_palindromic(s[i:j]) and len(s[i:j]) > max_len:
                    max_len = len(s[i:j])
                    ans = s[i:j]
        
        return ans

# Time Complexity: O(n^3)
# Runtime: Time Limit Exceeded

File: 5_Longest_Palindromic_Substring/brute_force.cpp
class Solution {
public:
    bool check_palindromic(string s){
        int n = s.size();
        for (int i = 0; i < n; i++){
            if (s[i] != s[n-1-i])
                return false;
        }
        return true;
    }

    string longestPalindrome(string s) {
        string ans = "";
        int max_len = 0;
        int n = s.size();

        for (int i = 0; i < n; i++){
            for (int len = 1; len <= n-i; len++){
                if (check_palindromic(s.substr(i, len)) && s.substr(i, len).size() > max_len){
                    max_len = s.substr(i, len).size();
                    ans = s.substr(i, len);
                }
            }
        }
        return ans;
    }
};


// Time Complexity: O(n^3)
// Runtime: Time Limit Exceeded